lib/release-ops.js

```diff
@@ -148,6 +148,58 @@ function getVersionTags() {
 // }
 
 /**
+ * Extracts data from a commit log in the format --pretty=format:"* %h %s (%an)".
+ * @param {string} log A single commit log.
+ * @returns {Object} An object containing the data exracted from the commit log.
+ * @private
+ */
+function parseLog(log) {
+    var regexp = /^(?:\* )?([0-9a-f]{7}) (?:(Revert) "((?:([a-z]+): ?)?.*)"|((?:([a-z]+): ?)?.*)) \((.*)\)/i,
+        match = log.match(regexp),
+        flag = match[6] || match[4];
+
+    return {
+        raw: match[0],
+        sha: match[1],
+        revert: !!match[2],
+        flag: flag ? flag.toLowerCase() : null,
+        title: match[5] || match[3],
+        author: match[7]
+    };
+}
+
+/**
+ * Given a list of parsed commit log messages, excludes revert commits and the
+ * commits they reverted.
+ * @param {Object[]} logs An array of parsed commit log messages.
+ * @returns {Object[]} An array of parsed commit log messages.
+ */
+function excludeReverts(logs) {
+    logs = logs.slice();
+
+    var titleIndexMap = Object.create(null), // Map of commit titles to indices
```

Comment:
Shouldn't we try to look at the body and get the actual SHA instead? Revert bodies usually look like `This reverts commit 86c5a20815b5246ce404700fa3921b4a06512b91.` which is easier to parse and gives you a more precise match because of the SHA.